qiskit_metal/toolbox_metal/layer_stack_handler.py

```diff
@@ -0,0 +1,439 @@
+                                        material=['pec'],
+                                        thickness=['2um'],
+                                        z_coord=['0um'],
+                                        fill=['false'])
+
+        self._init_dataframe()
+
+    def _init_dataframe(self) -> None:
+        """Must check if filename for layerstack is valid before trying to import to a pandas table.
+        """
+
+        if self.filename_csv_df:
+            #populate pandas table from data in this file.
+            abs_path = os.path.abspath(self.filename_csv_df)
+            if os.path.isfile(abs_path):
+                self._read_csv_df(abs_path)
+            else:
+                self.logger.error(
+                    f'Not able to read file.'
+                    f'File:{abs_path} not read. Check the name and path.')
+
+        else:
+            #enter very basic default data for pandas table.
+            self.ls_df = pd.DataFrame(data=self.layer_stack_default)
+
+    def is_layer_unique(self) -> bool:
+        """Check to sort on layer number make sure they are unique.
+        This method is not so relevant, since layers can have datatype entries.
+        Thus there can be rows with same layer number.
+        """
+        return self.ls_df['layer'].is_unique
+
+    def _read_csv_df(self, abs_path: str) -> None:
+        #ASSUME that self.filename_csv_df is valid file path and name.
+        #https://best-excel-tutorial.com/59-tips-and-tricks/485-how-to-display-a-single-quote-in-a-cell
+        try:
+            self.ls_df = pd.read_csv(abs_path)
+
+        except BaseException as error:
+            self.logger.warning(
+                f'Not able to create pandas dataframe.'
+                f'File:{abs_path} not imported. Expected a CSV formatted file.')
+            self.logger.error(f'The exception: {error}')
+
+    def get_fill_for_chip_layer_datatype(self,
+                                         chip_name: str,
+                                         layer_num: int,
+                                         datatype=0) -> Tuple[bool, bool]:
+        """Determine if geometry should be filled in Ansys.
+
+        Args:
+            chip_name (str): Name of chip which has the layer of interest.
+            layer_num (int): The unique layer number through out all chips.
+                            The same layer number can  not be used on multiple chips.
+            datatype (int, optional):  Datatype of the layer number. Defaults to 0.
+
+        Returns:
+            Tuple[bool, bool]: 1st bool denotes if value for fill was available,
+                            2nd value denotes value for fill. Default is None.
+        """
+        found_fill_value = False
+        fill_value = None
+        mask = (self.ls_df['layer']
+                == layer_num) & (self.ls_df['datatype'] == datatype) & (
+                    self.ls_df['chip_name'].str.contains(chip_name))
```

Comment:
This is interesting!! Didn't know this could be done with pandas tables! 🤩

Reply:
Thanks